esmvaltool/diag_scripts/examples/my_little_diagnostic.py

```diff
@@ -38,9 +49,45 @@ def _get_my_files(cfg):
     return files_dict
 
 
+def _run_diagnostic(cfg):
+    """
+    Simple example of a diagnostic.
+
+    This is a basic (and rather esotherical) diagnostic that firstly
+    loads the needed model data as iris cubes, performs a difference between
+    values at ground level and first vertical level, then squares the
+    result. The user will implement their own (custom) diagnostics, but this
+    example shows that once the preprocessor has finished a whole lot of
+    user-specific metrics can be computed as part of the diagnostic.
+
+    """
+    # assemble the data dictionary keyed by dataset name
+    my_files_dict = _get_my_files(cfg)
```

Comment:
The function `_get_my_files` can be replaced by `group_metadata(cfg['input_data'].values(), 'dataset')`